libs/mark-flow-ui/src/components/candidate_contest.tsx

```diff
@@ -87,8 +88,46 @@ const WriteInModalActionsSidebar = styled.div`
   justify-content: center;
 `;
 
-function findCandidateById(candidates: readonly Candidate[], id: string) {
-  return candidates.find((c) => c.id === id);
+/**
+ * Creates a unique choice identifier for a candidate option that includes
+ * party information to distinguish between multiple options for cross-endorsed
+ * candidates.
+ */
+function getCandidateChoiceId(
+  candidateId: CandidateId,
+  partyIds?: readonly string[]
+): string {
+  if (!partyIds || partyIds.length === 0) {
+    return candidateId;
+  }
+  // Sort party IDs to ensure consistent ordering
+  return `${candidateId}|${[...partyIds].sort().join(',')}`;
+}
```

Comment:
I'm a bit worried that using a string to identify unique candidate options will make it easy to make a mistake where we accidentally use the wrong id (e.g. using `candidate.id` instead). Would it be possible to just pass around the entire candidate object instead? Or maybe we could make a new object that represents an option and use that instead of candidate objects.

E.g.
```ts
type CandidateOption {
  candidate: Candidate,
  equals(otherOption: CandidateOption) {
     // compare candidate.id and candidate.partyIds
  }
}
```

I feel like this edge case is pretty complicated so I'm grasping for something to make it feel safer/more manageable at a type level.